templates/partial/search_result.html

```diff
@@ -7,7 +7,11 @@
       <div class="govuk-grid-column-full">
         <h3 class="govuk-heading-m govuk-!-margin-bottom-2">
           {% with result_type=result.result_type.name|lower %}
-            <a href="{% url 'home:details' result_type=result_type urn=result.urn %}" class="govuk-link">{{result.fully_qualified_name}}</a>
+            {% if result.display_name %}
+              <a href="{% url 'home:details' result_type=result_type urn=result.urn %}" class="govuk-link">{{result.display_name}}</a>
+            {% else %}
```

Comment:
`display_name` already defaults to `name` in `parse_names` so I *think* this will always display something, and you don't need the `else`

I think I prefer having all the name defaulting logic in the library itself rather than the templates